app/models/step.rb

```diff
@@ -90,6 +90,36 @@ def my_module
     protocol.present? ? protocol.my_module : nil
   end
 
+  def move_up!
+    return unless position.positive?
+
+    step_above = protocol.steps.find_by(position: position - 1)
+
+    return unless step_above
+
+    transaction do
+      step_above.position = position
+      update!(position: -1)
+      step_above.save!
+      update!(position: step_above.position - 1)
+    end
+  end
+
+  def move_down!
+    return unless position < protocol.steps.count - 1
+
+    step_below = protocol.steps.find_by(position: position + 1)
+
+    return unless step_below
+
+    transaction do
+      step_below.position = position
+      update!(position: -1)
+      step_below.save!
+      update!(position: step_below.position + 1)
```

Comment:
would this work?:
update([step_below.id, self.id], {position: position}, {position: step_below.position})